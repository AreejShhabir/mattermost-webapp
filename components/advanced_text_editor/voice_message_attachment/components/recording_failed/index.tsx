// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, useIntl} from 'react-intl';

import {MicrophoneIcon, CloseIcon} from '@mattermost/compass-icons/components';

import {AttachmentRootContainer,
    CancelButton,
    TextColumn,
    Title,
    Subtitle} from 'components/advanced_text_editor/voice_message_attachment/components/file_attachment_containers';

interface Props {
    onCancel: () => void;
}

const VoiceMessageRecordingFailed = (props: Props) => {
    const intl = useIntl();

    const errorMessage = intl.formatMessage({
        id: 'voiceMessage.recordingFailed',
        defaultMessage: 'Failed to start recording',
    });

    return (
        <AttachmentRootContainer
            icon={(
                <MicrophoneIcon
                    size={24}
                    color='#FFFFFF'
                />
            )}
            iconDanger={true}
        >
            <TextColumn>
                <Title>
                    <FormattedMessage
                        id='voiceMessage.preview.title'
                        defaultMessage='Voice message'
                    />
                </Title>
                <Subtitle title={errorMessage}>
                    {errorMessage}
                </Subtitle>
            </TextColumn>
            <CancelButton onClick={props.onCancel}>
                <CloseIcon
                    size={18}
                />
            </CancelButton>
        </AttachmentRootContainer>
    );
};

export default VoiceMessageRecordingFailed;

