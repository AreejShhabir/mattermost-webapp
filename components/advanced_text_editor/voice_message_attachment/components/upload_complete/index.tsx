// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import styled from 'styled-components';

import {
    CloseIcon,
    PlayIcon,
    PauseIcon,
} from '@mattermost/compass-icons/components';

import {Theme} from 'mattermost-redux/selectors/entities/preferences';

import {convertSecondsToMSS} from 'utils/datetime';

import {useAudioPlayer, AudioPlayerState} from 'components/common/hooks/useAudioPlayer';

import {AttachmentContainer, CancelButton, Duration} from '../containers';

interface Props {
    theme: Theme;
    src: string;
    onCancel: () => void;
}

const VoiceMessageUploadCompleted = (props: Props) => {
    const {playerState, duration, togglePlayPause} = useAudioPlayer(props.src);

    return (
        <AttachmentContainer
            icon={
                playerState === AudioPlayerState.Playing ? (
                    <PauseIcon
                        size={24}
                        color={props.theme.buttonBg}
                    />
                ) : (
                    <PlayIcon
                        size={24}
                        color={props.theme.buttonBg}
                    />
                )
            }
            onIconClick={togglePlayPause}
        >
            <VisualizerContainer><span>k</span></VisualizerContainer>
            <Duration>
                {convertSecondsToMSS(duration)}
            </Duration>
            <CancelButton onClick={props.onCancel}>
                <CloseIcon size={18}/>
            </CancelButton>
        </AttachmentContainer>
    );
};

export const VisualizerContainer = styled.div`
    flex-grow: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    padding-right: 1rem;
`;

export default VoiceMessageUploadCompleted;
