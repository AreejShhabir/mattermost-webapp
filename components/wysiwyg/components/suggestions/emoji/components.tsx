// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import styled from 'styled-components';

import RenderEmoji from 'components/emoji/render_emoji';

const MentionItemRoot = styled.div`
    display: flex;
    flex: 1;
    gap: 8px;
    padding: 8px 20px;
    align-items: center;

    font-family: 'Open Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 20px;

    color: rgb(var(--center-channel-color-rgb));

    > svg:first-child {
        margin: 3px;
        opacity: 0.56;
    }

    &:hover svg:first-child {
        opacity: 0.72;
    }

    .status-wrapper {
        height: auto;
    }
`;

const EmojiSuggestionItem = ({name}: {name: string}) => {
    return (
        <MentionItemRoot
            id={name}
            aria-label={name}
        >
            <RenderEmoji emojiName={name}/>
            {name}
        </MentionItemRoot>
    );
};

export {
    EmojiSuggestionItem,
};
