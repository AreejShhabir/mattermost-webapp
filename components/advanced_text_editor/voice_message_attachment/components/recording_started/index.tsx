// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {forwardRef} from 'react';
import {FormattedMessage} from 'react-intl';
import styled from 'styled-components';

import {MicrophoneIcon, CloseIcon, CheckIcon} from '@mattermost/compass-icons/components';

import {Theme} from 'mattermost-redux/selectors/entities/preferences';

import {convertSecondsToMSS} from 'utils/datetime';

import {AttachmentContainer, CancelButton, OkButton, Duration} from '../containers';

interface Props {
    theme: Theme;
    elapsedTime: number;
    onCancel: () => void;
    onComplete: () => Promise<void>;
}

const VoiceMessageRecordingStarted = forwardRef((props: Props, ref: React.Ref<HTMLCanvasElement>) => {
    return (
        <AttachmentContainer
            icon={(
                <MicrophoneIcon
                    size={24}
                    color={props.theme.buttonBg}
                />
            )}
        >
            <VisualizerContainer>
                <Canvas ref={ref}>
                    <FormattedMessage
                        id='voiceMessage.canvasFallback.recording'
                        defaultMessage='Recording started'
                    />
                </Canvas>
            </VisualizerContainer>
            <Duration>
                {convertSecondsToMSS(props.elapsedTime)}
            </Duration>
            <CancelButton onClick={props.onCancel}>
                <CloseIcon
                    size={18}
                />
            </CancelButton>
            <OkButton onClick={props.onComplete}>
                <CheckIcon
                    size={18}
                    color={props.theme.buttonColor}
                />
            </OkButton>
        </AttachmentContainer>
    );
});

export const VisualizerContainer = styled.div`
    flex-grow: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    padding-right: 1rem;
`;

const Canvas = styled.canvas`
    width: 100%;
    height: 24px;
`;

export default VoiceMessageRecordingStarted;

