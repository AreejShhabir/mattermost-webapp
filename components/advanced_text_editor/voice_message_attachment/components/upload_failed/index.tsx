// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, useIntl} from 'react-intl';

import {CloseIcon, RefreshIcon} from '@mattermost/compass-icons/components';

import {AttachmentContainer, CancelButton, TextColumn, Title, Subtitle} from '../containers';

interface Props {
    onRetry: () => void;
    onCancel: () => void;
}

const VoiceMessageRecordingFailed = (props: Props) => {
    const intl = useIntl();

    const errorMessage = intl.formatMessage({
        id: 'voiceMessage.uploadFailed',
        defaultMessage: 'Upload failed. Click to retry.',
    });

    return (
        <AttachmentContainer
            icon={(
                <RefreshIcon
                    size={24}
                    color='#FFFFFF'
                />
            )}
            iconDanger={true}
            onIconClick={props.onRetry}
        >
            <TextColumn>
                <Title>
                    <FormattedMessage
                        id='voiceMessage.preview.title'
                        defaultMessage='Voice message'
                    />
                </Title>
                <Subtitle title={errorMessage}>
                    {errorMessage}
                </Subtitle>
            </TextColumn>
            <CancelButton onClick={props.onCancel}>
                <CloseIcon
                    size={18}
                />
            </CancelButton>
        </AttachmentContainer>
    );
};

export default VoiceMessageRecordingFailed;

