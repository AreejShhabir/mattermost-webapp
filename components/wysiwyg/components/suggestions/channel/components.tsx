// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import styled from 'styled-components';
import {GlobeIcon} from '@mattermost/compass-icons/components';

import {Channel} from '@mattermost/types/channels';

const MentionItemRoot = styled.div`
    display: flex;
    flex: 1;
    gap: 8px;
    padding: 8px 20px;
    align-items: center;

    font-family: 'Open Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 20px;

    color: rgb(var(--center-channel-color-rgb));

    > svg:first-child {
        margin: 3px;
        opacity: 0.56;
    }

    &:hover svg:first-child {
        opacity: 0.72;
    }

    .status-wrapper {
        height: auto;
    }
`;

const MentionItemDescription = styled.span`
    opacity: 0.52;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
`;

const ChannelSuggestionItem = (channel: Channel) => {
    const {id, display_name: displayName, name} = channel;
    return (
        <MentionItemRoot
            id={id}
            aria-label={name}
        >
            <GlobeIcon size={18}/>
            {displayName}
            <MentionItemDescription>{`@${name}`}</MentionItemDescription>
        </MentionItemRoot>
    );
};

export {
    ChannelSuggestionItem,
};
