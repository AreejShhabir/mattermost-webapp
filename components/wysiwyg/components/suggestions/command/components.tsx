// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import styled from 'styled-components';
import {DockWindowIcon} from '@mattermost/compass-icons/components';

import Constants from 'utils/constants';

import Badge from 'components/widgets/badges/badge';

import {AutocompleteSuggestion} from '@mattermost/types/autocomplete';

const MentionItemRoot = styled.div`
    display: flex;
    flex: 1;
    gap: 8px;
    padding: 8px 20px;
    align-items: center;

    font-family: 'Open Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 20px;

    color: rgb(var(--center-channel-color-rgb));

    > svg:first-child {
        margin: 3px;
        opacity: 0.56;
    }

    &:hover svg:first-child {
        opacity: 0.72;
    }

    .status-wrapper {
        height: auto;
    }
`;

const MentionItemDescription = styled.span`
    opacity: 0.52;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
`;

const EXECUTE_CURRENT_COMMAND_ITEM_ID = Constants.Integrations.EXECUTE_CURRENT_COMMAND_ITEM_ID;
const OPEN_COMMAND_IN_MODAL_ITEM_ID = Constants.Integrations.OPEN_COMMAND_IN_MODAL_ITEM_ID;
const COMMAND_SUGGESTION_ERROR = Constants.Integrations.COMMAND_SUGGESTION_ERROR;

const CommandItem = (props: AutocompleteSuggestion) => {
    const {Suggestion, Hint, Description, IconData} = props;
    const label = `${Suggestion} ${Hint}`;

    let icon;
    switch (IconData) {
    case EXECUTE_CURRENT_COMMAND_ITEM_ID:
        icon = <span className='block mt-1'>{'â†µ'}</span>;
        break;
    case OPEN_COMMAND_IN_MODAL_ITEM_ID:
        icon = <DockWindowIcon size={28}/>;
        break;
    case COMMAND_SUGGESTION_ERROR:
        icon = <span>{'!'}</span>;
        break;
    }
    if (IconData && ![EXECUTE_CURRENT_COMMAND_ITEM_ID, COMMAND_SUGGESTION_ERROR, OPEN_COMMAND_IN_MODAL_ITEM_ID].includes(IconData)) {
        icon = (
            <div
                className='slash-command__icon'
                style={{backgroundColor: 'transparent'}}
            >
                <img src={IconData}/>
            </div>
        );
    }

    return (
        <MentionItemRoot
            id={`${Suggestion}_${Hint}`}
            aria-label={label}
        >
            {IconData ? icon : <Badge size={'lg'}>{' / '}</Badge>}
            {label}
            {Description && (<MentionItemDescription>{Description}</MentionItemDescription>)}
        </MentionItemRoot>
    );
};

export {
    CommandItem,
};
