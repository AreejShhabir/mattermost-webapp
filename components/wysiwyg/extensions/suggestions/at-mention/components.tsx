// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {AccountMultipleOutlineIcon} from '@mattermost/compass-icons/components';
import React from 'react';
import {useIntl} from 'react-intl';
import {useSelector} from 'react-redux';
import {getDirectTeammate} from 'mattermost-redux/selectors/entities/channels';
import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
import {getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
import {isGuest} from 'mattermost-redux/utils/user_utils';
import styled from 'styled-components';

import {getLongDisplayNameParts, imageURLForUser} from 'utils/utils';
import {GlobalState} from 'types/store';

import CustomStatusEmoji from 'components/custom_status/custom_status_emoji';
import ProfilePicture from 'components/profile_picture';
import BotBadge from 'components/widgets/badges/bot_badge';
import GuestBadge from 'components/widgets/badges/guest_badge';

import {Group} from '@mattermost/types/groups';

import {UserProfile} from '@mattermost/types/users';

const MentionItemRoot = styled.div`
    display: flex;
    flex: 1;
    gap: 8px;
    padding: 8px 20px;
    align-items: center;

    font-family: 'Open Sans', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 20px;

    color: rgb(var(--center-channel-color-rgb));

    > svg:first-child {
        margin: 3px;
        opacity: 0.56;
    }

    &:hover svg:first-child {
        opacity: 0.72;
    }

    .status-wrapper {
        height: auto;
    }
`;

const MentionItemDescription = styled.span`
    opacity: 0.52;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
`;

const UserMentionItem = (user: UserProfile) => {
    const {formatMessage} = useIntl();

    const currentUserId = useSelector(getCurrentUserId);
    const status = useSelector((state: GlobalState) => getStatusForUserId(state, user.id));
    const teammate = useSelector((state: GlobalState) => getDirectTeammate(state, user.id));
    const {displayName, fullName, nickname} = getLongDisplayNameParts(user);

    let name = displayName;

    if (user.id === currentUserId) {
        name += ` ${formatMessage({id: 'suggestion.user.isCurrent', defaultMessage: '(you)'})}`;
    }

    let description = fullName ?? '';

    if (nickname) {
        description += ` (${nickname})`;
    }

    if (user.delete_at) {
        description += ` - ${formatMessage({id: 'channel_switch_modal.deactivated', defaultMessage: 'Deactivated'})}`;
    }

    const tag = (
        <>
            <BotBadge
                show={Boolean(teammate?.is_bot)}
                className='badge-autocomplete'
            />
            <GuestBadge
                show={Boolean(teammate && isGuest(teammate.roles))}
                className='badge-autocomplete'
            />
        </>
    );

    const userImageUrl = imageURLForUser(user.id, user.last_picture_update);

    const emojiStyle = {
        marginBottom: 2,
    };

    return (
        <MentionItemRoot
            id={user.id}
            aria-label={name}
        >
            <ProfilePicture
                src={userImageUrl}
                status={teammate?.is_bot ? undefined : status}
                size='sm'
            />
            {name}
            {description && <MentionItemDescription>{description}</MentionItemDescription>}
            <CustomStatusEmoji
                showTooltip={true}
                userID={user.id}
                emojiStyle={emojiStyle}
            />
            {tag}
        </MentionItemRoot>
    );
};

const GroupMentionItem = (group: Group) => {
    const {id, description, name} = group;
    return (
        <MentionItemRoot
            id={id}
            aria-label={name}
        >
            <AccountMultipleOutlineIcon size={18}/>
            {`@${name}`}
            <MentionItemDescription>{description}</MentionItemDescription>
        </MentionItemRoot>
    );
};

const SpecialMentionItem = ({name}: {name: string}) => {
    const {formatMessage} = useIntl();
    const description = formatMessage({id: `suggestion.mention.${name}`});
    return (
        <MentionItemRoot
            id={`${name}_mention-item`}
            aria-label={name}
        >
            <AccountMultipleOutlineIcon size={18}/>
            {`@${name}`}
            <MentionItemDescription>{description}</MentionItemDescription>
        </MentionItemRoot>
    );
};

export {
    UserMentionItem,
    GroupMentionItem,
    SpecialMentionItem,
};
