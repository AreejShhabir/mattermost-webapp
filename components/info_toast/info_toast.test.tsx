// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {CheckIcon} from '@mattermost/compass-icons/components';

import InfoToast from './info_toast';

describe('components/InfoToast', () => {
    test('should match snapshot', () => {
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal: jest.fn()}}
                content={{message: 'A message'}}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should close the toast on undo', () => {
        const closeModal = jest.fn();
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal}}
                content={{message: 'A message', undo: jest.fn()}}
            />,
        );

        wrapper.find('button').simulate('click');

        expect(closeModal).toHaveBeenCalled();
    });

    test('should close the toast on timeout', () => {
        jest.useFakeTimers();

        const closeModal = jest.fn();
        shallow(
            <InfoToast
                actions={{closeModal}}
                content={{message: 'A message'}}
            />,
        );

        // setTimeout(() => {
        //     const activeSlide = wrapper.find('div.active-anim');
        //     const slide1Text = activeSlide.find('p.slide').text();
        //     expect(slide1Text).toEqual('Second Slide');
        //     done();
        // }, 1000);

        jest.runAllTimers();

        expect(closeModal).toHaveBeenCalled();
    });

    test('should close the toast on close button click', () => {
        const closeModal = jest.fn();
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal}}
                content={{message: 'A message'}}
            />,
        );

        wrapper.find('.info-toast__icon_button').simulate('click');

        expect(closeModal).toHaveBeenCalled();
    });

    test('should match snapshot with icon', () => {
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal: jest.fn()}}
                content={{message: 'A message', icon: <CheckIcon/>}}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with undo', () => {
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal: jest.fn()}}
                content={{message: 'A message', undo: jest.fn()}}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with className', () => {
        const wrapper = shallow(
            <InfoToast
                actions={{closeModal: jest.fn()}}
                content={{message: 'A message'}}
                className='className'
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
