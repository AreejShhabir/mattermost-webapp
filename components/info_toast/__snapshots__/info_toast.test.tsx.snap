// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/InfoToast shoul match snapshot with icon 1`] = `
<CSSTransition
  appear={true}
  classNames="slide"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="info-toast"
  >
    <div>
      icon
      <span>
        A message
      </span>
    </div>
    <ForwardRef
      className="info-toast__icon_button"
      icon="close"
      inverted={true}
      onClick={[Function]}
      size="sm"
    />
  </div>
</CSSTransition>
`;

exports[`components/InfoToast should match snapshot 1`] = `
<CSSTransition
  appear={true}
  classNames="slide"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="info-toast"
  >
    <div>
      <span>
        A message
      </span>
    </div>
    <ForwardRef
      className="info-toast__icon_button"
      icon="close"
      inverted={true}
      onClick={[Function]}
      size="sm"
    />
  </div>
</CSSTransition>
`;

exports[`components/InfoToast should match snapshot with className 1`] = `
<CSSTransition
  appear={true}
  classNames="slide"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="info-toast className"
  >
    <div>
      <span>
        A message
      </span>
    </div>
    <ForwardRef
      className="info-toast__icon_button"
      icon="close"
      inverted={true}
      onClick={[Function]}
      size="sm"
    />
  </div>
</CSSTransition>
`;

exports[`components/InfoToast should match snapshot with icon 1`] = `
<CSSTransition
  appear={true}
  classNames="slide"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="info-toast"
  >
    <div>
      <CheckIcon />
      <span>
        A message
      </span>
    </div>
    <ForwardRef
      className="info-toast__icon_button"
      icon="close"
      inverted={true}
      onClick={[Function]}
      size="sm"
    />
  </div>
</CSSTransition>
`;

exports[`components/InfoToast should match snapshot with undo 1`] = `
<CSSTransition
  appear={true}
  classNames="slide"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="info-toast"
  >
    <div>
      <span>
        A message
      </span>
      <button
        className="info-toast__undo"
        onClick={[Function]}
      >
        Undo
      </button>
    </div>
    <ForwardRef
      className="info-toast__icon_button"
      icon="close"
      inverted={true}
      onClick={[Function]}
      size="sm"
    />
  </div>
</CSSTransition>
`;
